src/main/java/com/sonyericsson/hudson/plugins/gerrit/trigger/gerritnotifier/ToGerritRunListener.java

```diff
@@ -328,18 +332,13 @@ protected void cleanUpGerritCauses(GerritCause firstFound, Run build) {
      * @param project the project that will be built.
      * @param event   the event that caused the build to be scheduled.
      */
-    public synchronized void onTriggered(Job project, GerritTriggeredEvent event) {
+    public void onTriggered(Job project, GerritTriggeredEvent event) {
         //TODO stop builds for earlier patch-sets on same change.
         memory.triggered(event, project);
         if (event instanceof GerritEventLifecycle) {
```

Comment:
This is a change that might have outside repercussions. The lifecycle listeners might have assumptions about owning the monitor when called _(which is normal behavior in most listener cases that I've seen)_
But I think I can live with that risk if this change is really needed.